### PR TITLE
Add mana requirements indicator to the mana pool recipes

### PR DESCRIPTION
**Pull request by socolio**
Oh... how to compare recipes with one pixel mana requirements... time to (test) fix
image

Tested on one of the latest nightly builds.
Only client-side changes (buts also tested on dedicated server for sabity level stability)

### CODE REVIEW

**Review by Alastors — CHANGES_REQUESTED** [review]
This is an intentional major design decision of Botania and violates our magic integrity policy

**Comment by socolio** [comment]
> I think it can be a nice option to see how much a recipe use instead of go to a wiki / view the code to find out how much it use. Sure we can think of a mechanic to unlock this with some magical research

Current mana bar is accurate enough to meet the needs of the players if mana consumprion is big enought. This change is targeting new players with low level of botania reserach.

**Comment by Alastors** [comment]
Honestly I'd make the argument that there isn't enough Botania integration to even think about this in that regards 🤷‍♂️

**Comment by socolio** [comment]
> Honestly I'd make the argument that there isn't enough Botania integration to even think about this in that regards 🤷‍♂️

In GTNH there exist some sets of recipes where mana consumption fluctuates so enought that it is hard to distinguish real values. of mana requirements

**Comment by socolio** [comment]
> I'm with Alastors on this one, this was explicitly designed this way by the original author so I don't feel privileged enough to change that.

There is literally no difference between 1000 and 700 mana requirements on low level Botania crafts (according to NEI).

**Comment by Alastors** [comment]
> > I'm with Alastors on this one, this was explicitly designed this way by the original author so I don't feel privileged enough to change that.
> 
> 
> 
> There is literally no difference between 1000 and 700 mana requirements on low level Botania crafts (according to NEI). That is bad.

🤷‍♂️ it's neither bad nor good, it is simply the mechanic.

**Comment by Alastors** [comment]
> This is an intentional major design decision of Botania and violates our magic integrity policy

Following up on this, just for case history and all of that, our policy is very explicit in its defense toward maintaining the artistic and mechanical design, here are the policies related to that. 

https://github.com/GTNewHorizons/GTNH-Dev-Doc/blob/master/developer's%20code%20of%20conduct.md

7. Regarding textures within GTNH
- i. Regarding magic textures: within the vision of the modpack, magic is meant to feel, look, and be different from GregTech, any form of GUI update for the sake of (or that would inadvertently bring) consistency with GT GUIs should be rejected.
- ii. Regarding new textures: try to keep textures consistent with the mod/mod group they're being added to.
- iii. Regarding tooltips: reduction is fine when needed, but if there is a specific consistency, try to stay within that.
- iv. Regarding inventory changing: GUIs that shift the colour of the player's internal inventory are allowed as long as its usage is consistent. Examples being: GT's Steam Machines, Magic Bee's Magic Apiary, Railcraft's Coke Oven, etc.

8. The design focus of Magic
As mentioned above, magic within GregTech: New Horizons is meant to have a strong, distinct look, feel, and execution to the typical tech mod in the pack. Thusly, it is of upmost importance that both the artistic and mechanical designs of each mod be treated with the upmost respect and taken into consideration in the creation of new magic content."

Likewise, I think the case history reflects negatively against changes like this: 
 https://github.com/GTNewHorizons/MagicBees/pull/32
 
Going further, despite all of the changes Vazkii made against what his original promises were, the lack of distinct numbers in botania was one of the few things that remained.
 
 # personal note
 
 I do not state this out of malice, nor in attempt to dissuade you from pursuing magic development in general. I am ecstatic to see people passionate about magic in general, and I desperately hope to see you continue this passion in the future. Thank you for your contribution regardless of whether it makes it into the pack or not.

File: src/main/java/vazkii/botania/client/integration/nei/recipe/RecipeHandlerManaPool.java [hunk 1]
```diff
@@ -86,17 +86,25 @@ public int recipiesPerPage() {
...
+		transferRects.add(new RecipeTransferRect(new Rectangle(70, 21, 18, 18), "botania.manaPool"));
 	}
 
 	@Override
 	public void drawBackground(int recipe) {
 		super.drawBackground(recipe);
+
+		int mana = ((CachedManaPoolRecipe) arecipes.get(recipe)).mana;
+        String manaRequired = Integer.toString(mana);
+        FontRenderer font = Minecraft.getMinecraft().fontRenderer;
```
Comment (combusterf) [inline src/main/java/vazkii/botania/client/integration/nei/recipe/RecipeHandlerManaPool.java:98]: Tab/spaces need to be fixed

**Comment by GedeonGrays** [comment]
Just a reminder, Alfheim (Botania addon) already implements numerical mana representation (config option). 
Maybe another reason to add Alfheim to GTNH ¯\_(ツ)_/¯

File: src/main/java/vazkii/botania/client/integration/nei/recipe/RecipeHandlerManaPool.java [hunk 2]
```diff
@@ -86,17 +86,25 @@ public int recipiesPerPage() {
...
+
 		GL11.glEnable(GL11.GL_BLEND);
 		GL11.glColor4f(1.0F, 1.0F, 1.0F, 0.5F);
 		GuiDraw.changeTexture(LibResources.GUI_MANA_INFUSION_OVERLAY);
-		GuiDraw.drawTexturedModalRect(45, 20, 38, 35, 92, 50);
-		HUDHandler.renderManaBar(32, 80, 0x0000FF, 0.75F, ((CachedManaPoolRecipe) arecipes.get(recipe)).mana, TilePool.MAX_MANA / 10);
+
+		GuiDraw.drawTexturedModalRect(45, 5, 38, 35, 92, 50);
+		HUDHandler.renderManaBar(32, 63, 0x0000FF, 0.75F, mana, TilePool.MAX_MANA / 10);
+		font.drawString(manaRequired, 84 - font.getStringWidth(manaRequired) / 2, 71, 0xFF03737F);
```
Comment (combusterf) [inline src/main/java/vazkii/botania/client/integration/nei/recipe/RecipeHandlerManaPool.java:106]: To be eligible, this needs a config option that defaults to not showing the numerical mana value

**Review by combusterf — CHANGES_REQUESTED** [review]

**Comment by chochem** [comment]
I am with alastor and combusterf as well. This is antithetical to the basic design principles of botania that makes people like the mod so much.

**Comment by Alastors** [comment]
image
Why don't we just port the book zoom feature, then we stay in line with the original intentions and design

**Comment by dvdmandt** [comment]
I'm torn. The inability to see exact numbers are indeed thematical and the original design and all that, and it's nice and immersive when you're just doing simple stuff by hand imho, but as soon as you want to really use it for something, you'll likely start looking at wikis, source code or spreadsheets to get the actual numbers anyway. I guess I'm not against the idea of adding some form of item or research (_botanical engineering - the science of mystery?_) or something to unlock this at least.

**Comment by Dream-Master** [comment]
lets do it like thaumcraft show the full recipe with values when you researched it.

**Comment by socolio** [comment]
PR may be rejected due to https://github.com/GTNewHorizons/Botania/pull/53

**Comment by Alastors** [comment]
Closed as per request of the PR owner

**Pull request closed without merging**